### PR TITLE
Added OpenTelemetry Initialization action for Dataproc clusters (along with integration tests)

### PR DESCRIPTION
**Pull request by akshatbhargava123**
- Original PR: #1072 
- Added integration test against OpenTelemetry init action
- Added support for rocky-8

PR credits: @shambhavirai16
Original author could not continue contribution.

### CODE REVIEW

**Comment by akshatbhargava123** [comment]
Added my official email in GH account, checking again if it works.

**Comment by jayadeep-jayaraman** [comment]
BUILD file is missing in the otel folder

```
2023-08-04T13:35:45.114355556Z ERROR: no such package 'otel': BUILD file not found in any of the following directories. Add a BUILD file to a directory to mark it as a package.
2023-08-04T13:35:45.114416017Z  - /init-actions/otel

```

**Comment by akshatbhargava123** [comment]
> BUILD file is missing in the otel folder
> 
> ```
> 2023-08-04T13:35:45.114355556Z ERROR: no such package 'otel': BUILD file not found in any of the following directories. Add a BUILD file to a directory to mark it as a package.
> 2023-08-04T13:35:45.114416017Z  - /init-actions/otel
> ```

Got it. I initially followed `cloud-sql-proxy` as an example where the BUILD instructions were specified in the project scoped (global) file. Updated it to have local BUILD file.

**Comment by akshatbhargava123** [comment]
Now only rocky-8 tests are failing due to some permission issue, it might be flaky so I'm re-running it.

File: otel/otel.sh [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+
+  # Storing the list of endpoints in a comma separated string
+  for element in "${endpoints_array[@]}"
+  do
+  prometheus_scrape_endpoints+="'$element',"
+  done
+
+  # Remove the extra comma if endpoints passed
+  length=${#prometheus_scrape_endpoints}
+  if [ $length -gt 0 ]; then
```
Comment (jayadeep-jayaraman) [inline otel/otel.sh:37]: can this be changed to `${length}`

Comment (akshatbhargava123) [reply otel/otel.sh]: Done.

**Review by jayadeep-jayaraman — APPROVED** [review]

**Pull request merged**